### DISCUSSION TITLE
ENOSPC: No space left on device

### DISCUSSION THREAD

**Discussion by insanity54** [windmill-labs/windmill#2729] [Questions] [answered]
I wrote a script (typescript/Bun) which

* downloads a file from URL to disk
* uploads the file to an IPFS cluster

I ran the script with a 1MB file and it works in my local windmill instance as well as windmill cloud. However when the file is larger, say 9GB, app.windmill.dev fails with an error saying that I've run out of disk space.

```
{
    "error": {
        "name": "ExecutionErr",
        "message": "ExitCode: 137, last log lines:\n3704 | }, Writable = require_writable(), _pathOrFdOrSink = Symbol(\"pathOrFdOrSink\"), _fileSink = Symbol(\"fileSink\"), _native = Symbol(\"native\");\n3705 | NativeWritable.prototype = Object.create(Writable.prototype);\n3706 | var WritablePrototypeWrite = Writable.prototype.write;\n3707 | \n3708 | NativeWritable.prototype.write = function NativeWritablePrototypeWrite(chunk, encoding, cb, native) {\n                                                                                                                                               ^\nENOSPC: No space left on device\n    fd: \"21\"\n syscall: \"write\"\n   errno: -28\n      at NativeWritablePrototypeWrite (node:stream:3708:140)\n      at node:stream:3331:122\n      at processTicksAndRejections (:61:76)"
    }
}
```

Is this something I can fix? Is there a way I can get a worker with a larger disk size?

**Comment by rubenfiszel** [comment, marked as answer]
HI @insanity54 ,

it's a limitation by design. Our multi-tenant offering has a very aggressive pricing and we cannot provide 10gb of disk per job for standard jobs. As a todo feature, we could implement more "expensive" jobs that could have more disk or more cpus but it's quite a bit of work at the moment. We're likely to implement that by offering premium worker groups with more cpu, more ram and more disk altogether.

At the moment the only way to use our cloud with more disk is to use our dedicated offerings that are quite a bit more expensive but you also get a dedicated cluster and everything.

Sorry about the limitation, will update if we bring more flexibility later on that front.

**Reply by insanity54** [reply]
Thanks @rubenfiszel. I'll self-host for now.
